drivers/SmartThings/sonos/src/lifecycle_handlers.lua

```diff
@@ -63,22 +67,35 @@ function SonosDriverLifecycleHandlers.initialize_device(driver, device)
           device.log.warn(string.format("error receiving device info: %s", recv_err))
         else
           ---@cast info { ssdp_info: SonosSSDPInfo, discovery_info: SonosDiscoveryInfo, force_refresh: boolean }
-          local success, error, error_code = driver:handle_player_discovery_info(driver:get_api_key(), info,
-            device)
-          if success then
-            return
+          local auth_success, api_key_or_err = driver:check_auth(info)
+          if not auth_success then
```

Comment:
This seems weird to have these 2 checks one after another one a "not" and the other == false

Reply:
The reason it's structured that way is because either the auth check can fail, which is a `false`, or the `check_auth` operation can error, which returns `nil, error`. In both cases, we want to call `device:offline()`, but in the `failed-but-not-error` case, we want to do the additional work in the `== false` branch.